nidirect_common/js/link-manager.js

```diff
@@ -0,0 +1,40 @@
+/**
+ * @file
+ * Defines Javascript behaviors for Link handling.
+ */
+(function ($) {
+
+  "use strict";
+
+  /**
+   * Changes any page links, within certain defined regions, from
+   * <a> elements to <span> elements if the current path matches the
+   * 'href' attribute.
+   *
+   * header.header
+   * #block-mainnavigation
+   * #main-content
+   * #footer
+   */
+  Drupal.behaviors.rewriteSelfReferencingLinks = {
+    attach: function (context, settings) {
+      const pathname = $(location).attr('pathname');
+      const elements = ['header.header', '#block-mainnavigation', '#main-content', '#footer'];
+
+      $.each(elements, function(index, elementRef) {
+        let classes = ['active', 'link__self'];
+
+        if (elementRef === '#block-mainnavigation') {
+          classes.push('nav-link');
+        }
+
+        $(elementRef + ' a[href*="' + pathname + '"]').each(function() {
+          if ($(this).attr('href') === pathname) {
```

Comment:
Not sure that this will work for relative paths, might it be worth thinking of a regex here ?

Reply:
Do you have an example for this? It's a somewhat crude matching process (contains) but `$(location).attr('pathname')` from http://example.com/article/a-title-value?arg1=1&arg2=2 yields `/article/a-title-value` so ought to suffice for the vast majority of cases.

I guess if we hit an edge case with something down the line, then it's a relatively simple thing to adjust.